src/js/components/TextInput/TextInput.js

```diff
@@ -448,6 +450,11 @@ const TextInput = forwardRef(
               readOnly
                 ? undefined
                 : event => {
+                    // when TextInput is not contained in a Form, no re-render
+                    // will come from this onChange and remove the placeholder,
+                    // so we need to update state to ensure the styled
+                    // placholder only appears when there is no value
+                    setShowStyledPlaceholder(!event.target.value);
```

Comment:
When I investigated this, it seems the onChange doesn't get called while the user is typing until they tab out or change focus. As a result the placeholder remains and the typed text overlaps it. Are you seeing something similar?

Reply:
onChange was getting called as soon as they were typing. However, it wasn't causing a rerender for whatever reason (which made the placeholder remain). I think it has to do with the fact that `value` is looking to formContext and so if the TextInput isn't in a formContext it's not registering the value change as a state change. 

by converting showStyledPlaceholder to something that is controlled by TextInput state, it allows us to toggle when it's shown by checking if the event.target has a value or not. Let me know if this makes sense.

Reply:
I was running this locally in Chrome. Let me confirm that it works across browsers.

Reply:
It is working as expected on Chrome, Firefox, and Safari.

Reply:
Looks great to me. I gave it a try and it definitely removes the placeholder on typing.